File: trace-agent-bootstrap/src/test/java/com/vmlens/trace/agent/bootstrap/parallelize/testFixture/ThreadModel.java
package com.vmlens.trace.agent.bootstrap.parallelize.testFixture;

public class ThreadModel {

    public final SyncActionModel[] syncActionModelArray;

    public ThreadModel(SyncActionModel[] syncActionModelArray) {
        this.syncActionModelArray = syncActionModelArray;
    }

    public static ThreadModel t(SyncActionModel... syncActionModels) {
        return new ThreadModel(syncActionModels);
    }

}

File: sync-bug/src/main/scala/com/vmlens/preanalyzed/model/classmodel/ThreadModel.scala
package com.vmlens.preanalyzed.model.classmodel

import com.vmlens.preanalyzed.model.ClassModel
import com.vmlens.trace.agent.bootstrap.preanalyzed.model.classtypeimpl.PreAnalyzedSpecificMethods
import com.vmlens.trace.agent.bootstrap.preanalyzed.model.methodtypeimpl.{ThreadJoin, ThreadStart}
import com.vmlens.trace.agent.bootstrap.preanalyzed.model.{PackageOrClass, PreAnalyzedMethod}

case class ThreadModel(name : String) extends ClassModel {
  def take(className: String) : Boolean = className.equals(name);

  override def create(): PackageOrClass = new PackageOrClass(name,
    PreAnalyzedSpecificMethods.SINGLETON,
    Array(new PreAnalyzedMethod("start", "()V", ThreadStart.SINGLETON),
      new PreAnalyzedMethod("join", "()V", ThreadJoin.SINGLETON)));
}
